tests/views_integration/README.md

```diff
@@ -316,10 +316,155 @@ This second test example shows what is, probably, the main feature of data-drive
+
+		$context = tribe_context()->alter(
+			[
+				'today'      => $this->mock_date_value,
+				'now'        => $this->mock_date_value,
+			]
+		);
+
+		$list_view = View::make( List_View::class, $context );
+		$html      = $list_view->get_html();
+
+		// Let's make sure the View is displaying what events we expect it to display.
+		$expected_post_ids = [];
+		$this->assertEquals(
+			$expected_post_ids,
+			$list_view->found_post_ids()
+		);
+
+		$this->assertMatchesSnapshot( $html );
+	}
+
+	/**
+	 * Test render with upcoming events
+	 */
+	public function test_render_with_upcoming_events() {
+		$events = [];
+
+		// Create the events.
+		foreach (
+			[
+				'tomorrow 9am',
+				'+1 week',
+				'+9 days',
+			] as $start_date
+		) {
+			$events[] = tribe_events()->set_args( [
+				'start_date' => $start_date,
+				'timezone'   => 'Europe/Paris',
+				'duration'   => 3 * HOUR_IN_SECONDS,
+				'title'      => 'Test Event - ' . $start_date,
+				'status'     => 'publish',
+			] )->create();
+		}
+		// Sanity check
+		$this->assertEquals( 3, tribe_events()->where( 'ends_after', 'now' )->count() );
+
+		$this->remap_posts( $events, [
+			'events/featured/1.json',
+			'events/single/1.json',
+			'events/single/2.json'
+		] );
+
+		$list_view = View::make( List_View::class );
+		$list_view->set_context( tribe_context()->alter( [
+			'today'      => $this->mock_date_value,
+			'now'        => $this->mock_date_value,
+			'events_per_page' => 2,
+		] ) );
+		$html = $list_view->get_html();
+
+		// Let's make sure the View is displaying what events we expect it to display.
+		$expected_post_ids = wp_list_pluck( array_slice( $events, 0, 2 ), 'ID' );
+		$this->assertEquals(
+			$expected_post_ids,
+			$list_view->found_post_ids()
+		);
+
+		$this->assertMatchesSnapshot( $html );
+	}
+}
+```
+
+By using the `MatchesSnapshots` trait and calling the `assertMatchesSnapshot` method, we can set an initial snapshot and compare the markup each time the test is run.
+
+When a markup change occurs, the test will fail as the html markup will not match the snapshot. In this case, review the differences. if they are what you expect, then delete the snapshot file and run the test again to generate a new snapshot. Commit this snapshot to the repo so that all others running tests will have the latest snapshot to compare to.
 
 ### Component (HTML) Testing
-@todo @paulmskim
+
+This type of testing deals with the following question:
+
+> Is this View partial rendering the correct markup (HTML structure, attributes, and data output) given a specific set of template variables?
+
+This testing is often run using snapshot testing as we are most interested in the View partial markup. See **Snapshot Testing** above for details.
+
+The View partials folder structure is organized in the same structure as the view partials. When creating a new test for a partial, place the test within the appropriate folder matching that of the actual markup.
+
+Below is an example of List View Nav:
```

Comment:
@paulmskim Above is said that this kind of testing "is often run using snapshot testing" and that's the only available example of testing. Please add more examples not using snapshot testing.